### ISSUE TITLE
[Bitbucket Cloud OAuth 2.0] ValueError:  rsa_key is required when using RSA signature method !

### ISSUE THREAD

**Issue by zineb-lotfi** [atlassian-api/atlassian-python-api#620]
1. here is how i configured my  bitbucket client but i dont provide key_cert

```
    def get_client(self):
        try:
            bitbucket_token = SocialToken.objects.get(
                account__user=self.user,
                account__provider='bitbucket_oauth2'
            )
            bitbucket_app = SocialApp.objects.get(provider='bitbucket_oauth2')
            oauth_dict = {
                'access_token': bitbucket_token.token,
                'access_token_secret': bitbucket_token.token_secret,
                'consumer_key': bitbucket_app.client_id,
                'key_cert': ''
            }
        except ObjectDoesNotExist:
            raise Exception(
                "User does not have a Bitbucket OAuth token"
            )
        try:
            client = Bitbucket(
                url=settings.BITBUCKET_CLOUD_API_URL,
                cloud=True,
                api_version='2.0',
                oauth=oauth_dict
            )
        except Exception as exc:
            raise Exception(
                "Failed to connect to the Bitbucket API {0}".format(exc)
            )
        return client
```
2. by doing by example:
  repos = self.client.repo_list(project_key='lorem test')

2. Expected Behavior:
  - return list of repos.

4. Actual behavior:
 - Error message ->  ValueError: rsa_key is required when using RSA signature method

i'm looking for what i did wrong or this is a real issue ?! help

**Comment by gonchik** [comment]
Hi!
How about to get token and then just put into password field

**Comment by WriteCodeEveryday** [comment]
> Hi!
> How about to get token and then just put into password field

https://github.com/atlassian-api/atlassian-python-api/issues/633

API Token for Password isn't working (for me)

**Comment by jensvog** [comment]
The current oauth implementation cannot be used for the cloud services. I think it was initially designed for the server variants. Cloud services nowadays use OAuth 2.0.

I implemented an approach for OAuth 2.0 with PR https://github.com/atlassian-api/atlassian-python-api/pull/666. There I also added an example Flask server where you can see how it is working. Then you don't need the rsa_key.

If you prefer using app passwords, you can take a look at my comment in issue https://github.com/atlassian-api/atlassian-python-api/issues/633

**Comment by gonchik** [comment]
@jensvog  Thank you for that PR.
